src/serde_impl/date.rs

```diff
@@ -0,0 +1,26 @@
+use chrono::{DateTime, TimeZone, Utc};
+use serde::{Deserializer, Serializer};
+use serde::de::{Error, Unexpected};
+use serde_with::{DeserializeAs, SerializeAs};
+use serde_with::json::JsonString;
+
+static FORMAT: &'static str = "%Y-%m-%d";
+
+pub(crate) fn deserialize<'de, D>(deserializer: D) -> Result<DateTime<Utc>, <D as Deserializer<'de>>::Error> where
+    D: Deserializer<'de> {
+    let date: String = JsonString::deserialize_as(deserializer)?;
+    Utc
+        .datetime_from_str(&date, FORMAT)
+        .ok()
+        .ok_or(D::Error::invalid_value(
+            Unexpected::Str(&date),
+            &"Expected a valid date string",
+        ))
+}
+
+pub(crate) fn serialize<S>(time: &DateTime<Utc>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer {
+    let date: String = time.format(FORMAT).to_string();
+    JsonString::serialize_as(&date, serializer)
```

Comment:
`JsonString` is used to interpret a string as JSON. Let's say i.e. You get
```json
{
  "some_field": "{ \"another_field\": null }"
}
```
form an API back. In that case, you would use JsonString to deserialize the content of `some_field`.
Deserialializing a `JsonString` into a `String` would mean something like `"\"2021-04-11\""`, so a `String` inside a `String`.

In this case, you can just use `String::deserialize`, or even `<&str>::deserialize` to eliminate an allocation (I checked the YouTube response) (same with deserialize).

Nevertheless, you will sooner or later interact with a part of the YouTube API, where a `JsonString` is actually needed. In these cases, I recommend using a `serde_as` attribute from the `serde_with` crate instead of hardcoding it into the serde implementation since it makes the implementation more generic, and other programmers instantly see, that this field is a `String` interpreted as JSON:

```rust
#[serde_as]
#[derive(Deserialize, Serialize)]
struct S {
  #[serde_as(as = "JsonString")]
  filed: AnotherStruct
}
```

I just checked my own serde implementations and noticed that I did something similar with `unix_timestamp_micro_secs` and `unix_timestamp_secs`. So it's my fault to give a bad example. I will change that in a future commit.

Reply:
I previously adopted your implementation of `unix_timestamp_micro_secs`. Now I implemented the solution from [the serde documentation](https://serde.rs/custom-date-format.html).